collectors/python.d.plugin/vnstat/vnstat.chart.py

```diff
@@ -0,0 +1,731 @@
+                not (interface["name"] in allowed_interfaces)
+            ):
+                continue
+
+            interfaces.append(interface["name"])
+
+            traffic = interface["traffic"]
+
+            total_rx = traffic["total"]["rx"]
+            total_tx = traffic["total"]["tx"]
+
+            current_amount = (
+                traffic["fiveminute"][(len(traffic["fiveminute"]) - 1)]["rx"]
+                + traffic["fiveminute"][(len(traffic["fiveminute"]) - 1)]["tx"]
+            )
+            current_data_rate = ((current_amount * 8) / 1000) / 300
+            # If no data is present for yesterday, then today's values will be chosen automatically:
+            yesterdayAmount = (
+                traffic["day"][(len(traffic["day"]) - 2)]["rx"]
+                + traffic["day"][(len(traffic["day"]) - 2)]["tx"]
+            )
+            yesterday_data_rate = ((yesterdayAmount * 8) / 1000) / 86400
+
+            hours_total_data = []
+            for hour in traffic["hour"]:
+                date = (
+                    str(hour["date"]["day"])
+                    + "/"
+                    + str(hour["date"]["month"])
+                    + "/"
+                    + str(hour["date"]["year"])
+                )
+                minute = str(hour["time"]["minute"])
+                if len(minute) == 1:
+                    minute = "0" + minute
+                time = str(hour["time"]["hour"]) + ":" + minute
+                hour_total = hour["rx"] + hour["tx"]
+                hours_total_data.append([(time + " " + date), hour_total])
+
+            days_total_data = []
+            for day in traffic["day"]:
+                date = (
+                    str(day["date"]["day"])
+                    + "/"
+                    + str(day["date"]["month"])
+                    + "/"
+                    + str(day["date"]["year"])
+                )
+                day_total = day["rx"] + day["tx"]
+                days_total_data.append([date, day_total])
+
+            months_total_data = []
+            for month in traffic["month"]:
+                date = (
+                    str(month["date"]["month"]) + "/" + str(month["date"]["year"])
+                )
+                month_total = month["rx"] + month["tx"]
+                months_total_data.append([date, month_total])
+
+            years_total_data = []
+            for year in traffic["year"]:
+                date = str(year["date"]["year"])
+                year_total = year["rx"] + year["tx"]
+                years_total_data.append([date, year_total])
+
+            this_interface_data = [
+                total_rx,
+                total_tx,
+                current_data_rate,
+                yesterday_data_rate,
+                hours_total_data,
+                days_total_data,
+                months_total_data,
+                years_total_data,
+            ]
+
+            data_points.append(this_interface_data)
```

Comment:
`get_data` could be split int `calc_hourly_traffic`, `calc_daily_traffic`, etc.

Reply:
True, although there would only be 3-5 lines of code per function... so it may not be worth it to split it up, as `_get_data` isn't too large, and is pretty well segmented